src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -770,5 +778,33 @@ private void HandleCustomHostLaunch(ITestHostLauncher customHostLauncher, Messag
                 this.communicationManager.SendMessage(MessageType.CustomTestHostLaunchCallback, ackPayload, this.protocolVersion);
             }
         }
+
+        private void AttachDebuggerToProcess(ITestHostLauncher customHostLauncher, Message message)
+        {
+            if (customHostLauncher == null)
+            {
+                throw new ArgumentNullException("VsTestConsoleRequestSender.AttachDebuggerToProcess: test host launcher cannot be null.");
+            }
+            if (!(customHostLauncher is IDebugTestHostLauncher launcher))
+            {
+                throw new NotSupportedException("VsTestConsoleRequestSender.AttachDebuggerToProcess: test host launcher is not ITestHostLauncher2.");
+            }
```

Comment:
id `customHostLauncher` is `null`, looks like the above `HandleCustomHostLaunch` actually returns -1 as response. Should we follow the same pattern here and return false for these cases instead of throwing?

Reply:
After removing the exceptions we do this.
By default the response is set to false and if the `customHostLauncher` is null we'll log an error message in the catch block. Either way in the finally block we send back the value of the response variable.